pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/ColumnMetadata.java

```diff
@@ -157,9 +164,27 @@ public static ColumnMetadata fromPropertiesConfiguration(String column, Properti
       }
     }
 
+
+    String partitionFunctionString = config.getString(getKeyFor(column, V1Constants.MetadataKeys.Column.PARTITION_FUNCTION));
```

Comment:
Partitioning is a segment property with mathematical constraint applied on a set of columns. Putting this on column metadata structurally impacts composite partitioning keys.

The property should be segment.partitioning.columns=a,b,c. There are plenty of examples for that.